test/distributed/tensor/test_utils.py

```diff
@@ -141,6 +142,52 @@ def _compute_start_end_offsets(self, global_offset, local_size, n_dim):
             offset.append(((global_offset[i]), (global_offset[i] + local_size[i])))
         return offset
 
+    @with_comms
+    def test_compute_global_tensor_shape_1D(self):
+        one_d_placements = [[Shard(1)], [Shard(0)], [Replicate()]]
+        device_mesh = init_device_mesh(self.device_type, (self.world_size,))
+        for placements in one_d_placements:
+            if isinstance(placements[0], Shard):
+                uneven_dim = list(range(self.world_size))
+                local_shape = (
+                    torch.Size([5, uneven_dim[device_mesh.get_rank()]])
+                    if placements[0].dim == 1
+                    else torch.Size([uneven_dim[device_mesh.get_rank()], 5])
+                )
+                expected_global_shape = (
+                    torch.Size([5, sum(uneven_dim)])
+                    if placements[0].dim == 1
+                    else torch.Size([sum(uneven_dim), 5])
+                )
+                global_shape = compute_global_tensor_shape(
+                    local_shape,
+                    device_mesh,
+                    placements,
+                )
+                self.assertEqual(
+                    global_shape,
+                    expected_global_shape,
+                )
+            else:
+                global_shape = compute_global_tensor_shape(
+                    torch.Size([5, 5]), device_mesh, placements
+                )
+                self.assertEqual(global_shape, torch.Size([5, 5]))
```

Comment:
this common piece of code can be moved out of the `if-else` with minor change. We can just calculate `local_shape` and `expected` in the branch.